linkerd/app/outbound/src/tcp/opaque_transport.rs

```diff
@@ -162,6 +167,32 @@ mod test {
         )
     }
 
+    fn http_ep(metadata: Metadata, version: http::Version) -> crate::http::Connect {
+        let e = crate::http::Endpoint::from((version, ep(metadata)));
+        crate::http::Connect::from((version, e))
+    }
```

Comment:
This stuff is going to conflict with other work, especially. Can you decouple the tests from the specific target types used in the stack?

Reply:
Okay, 755767627b7b3894a96b0c29b654d3c014f2c4ea changes these tests so that they no longer use the actual target types --- is that sufficient?

IMO, this is somewhat unfortunate, as the target types' `Param` impls were part of the code being tested here, and now, we need to ensure that the targets used in the tests actually behave the same as the real targets. But, hopefully that reduces the conflict with other changes?

Reply:
Look at what's coming in 
https://github.com/linkerd/linkerd2-proxy/pull/2210/files#diff-397930556d70c6560dae17b17cf26153f2b1398664e6e7b46112d5135a1ec4bb -- we don't have any reason to use `Metadata` at all.

> IMO, this is somewhat unfortunate, as the target types' Param impls were part of the code being tested here,

This decoupling is intended to force this conversation. Why are these tests coupled to the caller context? Do we need tests in the caller module(s) that exercise how they set their parameters? Or does this indicate that the params are not factored properly so that the inputs to this stack are too fine grain?